### ISSUE TITLE
False positive used-before-assignment with try-except

### ISSUE THREAD

**Issue by TTKuLa** [PyCQA/pylint#5817]
### Bug description

The following code will raise a `used-before-assignment` error:

```
def some_method():
    try:
        some_list = [e for e in (1, 2, 3) if e]  # Using variable 'e' before assignment!
    except Exception as e:
        pass
```

### Command used

```shell
pylint used_before_assignment.py -E
```

### Pylint output

```shell
************* Module used_before_assignment
used_before_assignment.py:3:45: E0601: Using variable 'e' before assignment (used-before-assignment)
```

### Expected behavior

Expected no error, since the variable `e` in the try-except and the `e` in the list comprehension use different scopes.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
Probably fixed in https://github.com/PyCQA/pylint/pull/5812 (duplicate of #5112 ?)

**Comment by DanielNoord** [comment]
I checked. Still emitting on `main` 😢

**Comment by jacobtylerwalls** [comment]
No, it's unrelated! Pretty much the only except-related case in that method that I had not altered yet :-D
